newIDE/app/src/EventsSheet/SearchPanel.js

```diff
@@ -33,7 +33,7 @@ type State = {|
   searchInConditions: boolean,
   searchInEventStrings: boolean,
   searchInSelection: boolean,
-  searchCount: boolean,
+  isSearchCountOne: boolean,
```

Comment:
This name is a bit better :) 
I would recommend to see it from the perspective of the search results:
Usually when we want to track if something was changed, we use what is called a "dirty flag".
Here I would use a flag called "searchResultsDirty".
it means: "when I'm true, the search results are 'dirty', i.e: they are not up to date anymore". In other words, a new search must be done.

Meaning that:
1. When search text is changed, the dirty flag is true
2. When the search is done, the dirty flag is false
3. When the events are changed, the dirty flag must be set to true again

So I would recommend to rename this to `searchResultsDirty` and invert the usage of true/false :)  
For 3., this means that the EventsSheet should call the search panel to tell to it "hey the events were modified". It can do this by using `this._searchPanel` and calling a function "markSearchResultsAsDirty" - which itself is just modifying the booleans.